pkg/deploy/deploy.go

```diff
@@ -63,6 +65,7 @@ func New(ctx context.Context, log *logrus.Entry, config *RPConfig, version strin
 	return &deployer{
 		log: log,
 
+		alertrules:        mgmtmonitor.NewAlertRulesClient(config.SubscriptionID),
```

Comment:
I don't see how this can work if it doesn't reference an authorizer.  I'd expect to see a new package under + github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt + mock generation + a constructor which takes an authorizer.

Reply:
Also it was the wrong API. The correct one to use was MetricAlerts not AlertRules.
Added authorizer in new azureclient + mocks.